pysnn/neuron.py

```diff
@@ -137,8 +137,11 @@ def refrac(self, spikes):
 
         Can be overwritten in case of the need of more refined functionality.
         """
-        self.refrac_counts[self.refrac_counts > 0] -= self.dt
-        self.refrac_counts += self.duration_refrac * self.convert_spikes(spikes)
+        if self.duration_refrac > 1:
+            self.refrac_counts[self.refrac_counts > 0] -= self.dt
+            self.refrac_counts += self.duration_refrac * self.convert_spikes(spikes)
+        else:
+            self.refrac_counts.copy_(self.convert_spikes(spikes))
```

Comment:
Wouldn't it be possible to just fill refrac count with 0 at initialization and never touch it again in case of no refrac?

Reply:
Yes, but that would change the way voltage is increased. Currently, setting `duration_refrac = 0` still means neurons can't spike in consecutive steps, because we do `duration_refrac + 1` in the `BaseNeuron` initialization. This in turn leads to the voltage for that neuron not being increased in the next timestep, so no spiking. So I think the refraction is currently implemented incorrectly, and I think removing the +1 would solve that (and then we can just leave the `self.refrac_counts` in that case)